File: typescript/src/state/concrete-state-b.ts
import { ConcreteStateA } from "./concrete-state-a.ts";
import type { Context } from "./context.ts";
import type { State } from "./state.ts";

export class ConcreteStateB implements State {
  handle(context: Context): void {
    context.setState(new ConcreteStateA());
  }

  name(): string {
    return "ConcreteStateB";
  }
}

File: python/src/patterns/state/concrete_state_b.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .context import Context


class ConcreteStateB:
    def handle(self, context: Context) -> None:
        # Local import: the two concrete states transition to each other.
        from .concrete_state_a import ConcreteStateA

        context.set_state(ConcreteStateA())

    def name(self) -> str:
        return "ConcreteStateB"
